src/main/java/com/zendesk/maxwell/schema/ddl/DatabaseDrop.java

```diff
@@ -0,0 +1,28 @@
+package com.zendesk.maxwell.schema.ddl;
+
+import com.zendesk.maxwell.schema.Database;
+import com.zendesk.maxwell.schema.Schema;
+
+public class DatabaseDrop extends SchemaChange {
+	public String dbName;
+	public boolean ifExists;
+
+	public DatabaseDrop(String dbName, boolean ifExists) {
+		this.dbName = dbName;
+		this.ifExists = ifExists;
+	}
+
+	@Override
+	public Schema apply(Schema originalSchema) throws SchemaSyncError {
+		Schema newSchema = originalSchema.copy();
+		Database database = findDatabase(newSchema, dbName, ifExists);
+
+		if ( database == null && !ifExists ) {
+			throw new SchemaSyncError("Can't drop missing database: " + dbName);
```

Comment:
i think you're right and that I should sacrifice a few lines of code for increased clarity and code locality.  will re-do the findDatabase and findTable stuff.